overlord/snapstate/snapmgr.go

```diff
@@ -51,11 +51,24 @@ import (
 )
 
 var (
-	snapdTransitionDelayWithRandomess = 3*time.Hour + randutil.RandomDuration(4*time.Hour)
+	ssPrng                            *randutil.PseudoRand
+	snapdTransitionDelayWithRandomess time.Duration
+
+	// overridden in the tests
+	errtrackerReport = errtracker.Report
 )
 
-// overridden in the tests
-var errtrackerReport = errtracker.Report
+func init() {
+	// Snapstate operations can impact server load so we need to use a seed
+	// that is not only based on time and PID, but contains
+	// additional variance introduced by device specific details such
+	// as hostname and MAC.
+	ssPrng = randutil.NewPseudoRand(randutil.SeedDatePidHostMac())
+
+	initAfterPrng()
```

Comment:
this is not great as it doesn't scale if we need to do more of this

Reply:
there might be a way to use the rules here https://go.dev/ref/spec#Package_initialization to avoid some of this